vllm/core/embedding_model_block_manager.py

```diff
@@ -0,0 +1,84 @@
+from typing import Dict, List
+
+from vllm.core.interfaces import AllocStatus, BlockSpaceManager
+from vllm.sequence import Sequence, SequenceGroup
+
+
+class EmbeddingModelBlockSpaceManager(BlockSpaceManager):
```

Comment:
The interfaces here are actively called in `Scheduler.schedule`. If we raise ValueError it would interrupt the schedule function.  I think it would make more sense to separate the schedule logic of embedding as it is pretty straightforward without block tables.